From 616581ba45c8d72645a29019702353f57dd0870d Mon Sep 17 00:00:00 2001
From: Thomas Lukaszewicz <tluk@chromium.org>
Date: Mon, 02 Feb 2026 16:41:26 -0800
Subject: [PATCH] [Vertical Tabs] Eliminate TabStripRegionView::StopAnimating()

API is no longer necessary. Existing clients are transitioned to
use ScopedAnimationDurationScaleMode instead.

Bug: 459824840
Change-Id: Idf5ac52b3a9128515af0e72c80885b270d28d55c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7534358
Reviewed-by: David Pennington <dpenning@chromium.org>
Commit-Queue: Tom Lukaszewicz <tluk@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1578463}
---

--- a/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
@@ -559,14 +559,6 @@ bool HorizontalTabStripRegionView::IsTab
   return tab_strip_->IsTabStripCloseable();
 }
 
-bool HorizontalTabStripRegionView::IsAnimating() const {
-  return tab_strip_->IsAnimatingInTabStrip();
-}
-
-void HorizontalTabStripRegionView::StopAnimating() {
-  return tab_strip_->StopAnimating();
-}
-
 void HorizontalTabStripRegionView::UpdateLoadingAnimations(
     const base::TimeDelta& elapsed_time) {
   tab_strip_->UpdateLoadingAnimations(elapsed_time);
--- a/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.h
+++ b/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.h
@@ -105,8 +105,6 @@ class HorizontalTabStripRegionView final
   bool IsTabStripEditable() const override;
   void DisableTabStripEditingForTesting() const override;
   bool IsTabStripCloseable() const override;
-  bool IsAnimating() const override;
-  void StopAnimating() override;
   void UpdateLoadingAnimations(const base::TimeDelta& elapsed_time) override;
   std::optional<int> GetFocusedTabIndex() const override;
   const TabRendererData& GetTabRendererData(int tab_index) override;
--- a/chrome/browser/ui/views/frame/tab_strip_region_view.h
+++ b/chrome/browser/ui/views/frame/tab_strip_region_view.h
@@ -34,8 +34,6 @@ class TabStripRegionView : public views:
   virtual bool IsTabStripEditable() const = 0;
   virtual void DisableTabStripEditingForTesting() const = 0;
   virtual bool IsTabStripCloseable() const = 0;
-  virtual bool IsAnimating() const = 0;
-  virtual void StopAnimating() = 0;
   virtual void UpdateLoadingAnimations(const base::TimeDelta& elapsed_time) = 0;
   virtual std::optional<int> GetFocusedTabIndex() const = 0;
   virtual const TabRendererData& GetTabRendererData(int tab_index) = 0;
--- a/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
@@ -222,14 +222,6 @@ bool VerticalTabStripRegionView::IsTabSt
   return !drag_controller || drag_controller->IsMovingLastTab();
 }
 
-bool VerticalTabStripRegionView::IsAnimating() const {
-  return false;
-}
-
-void VerticalTabStripRegionView::StopAnimating() {
-  // Do nothing.
-}
-
 void VerticalTabStripRegionView::UpdateLoadingAnimations(
     const base::TimeDelta& elapsed_time) {
   for (tabs::TabInterface* tab : *tab_strip_model_) {
--- a/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.h
+++ b/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.h
@@ -105,8 +105,6 @@ class VerticalTabStripRegionView final :
   bool IsTabStripEditable() const override;
   void DisableTabStripEditingForTesting() const override;
   bool IsTabStripCloseable() const override;
-  bool IsAnimating() const override;
-  void StopAnimating() override;
   void UpdateLoadingAnimations(const base::TimeDelta& elapsed_time) override;
   std::optional<int> GetFocusedTabIndex() const override;
   const TabRendererData& GetTabRendererData(int tab_index) override;
